app/com/linkedin/drelephant/ElephantRunner.java

```diff
@@ -178,8 +178,11 @@ public void run() {
         result.save();
         long processingTime = System.currentTimeMillis() - analysisStartTimeMillis;
         logger.info(String.format("Analysis of %s took %sms", analysisName, processingTime));
-        MetricsController.setJobProcessingTime(processingTime);
-        MetricsController.markProcessedJobs();
+
+        if(MetricsController.isInitialized()) {
```

Comment:
We should move this check inside the MetricsController. Methods like _setJobProcessingTime_ and _markProcessedJobs_ should check if MetricsController is initialized.